src/pydvl/value/shapley/classwise.py

```diff
@@ -0,0 +1,301 @@
+"""
+Refs:
+
+[1] CS-Shapley: Class-wise Shapley Values for Data Valuation in Classification (https://arxiv.org/abs/2211.06800)
+"""
+
+# TODO rename function, and object refs, when transferred to pyDVL.
+
+import numbers
+import operator
+from functools import reduce
+from typing import Optional, Sequence, Tuple
+
+import numpy as np
+from numpy._typing import NDArray
+
+from pydvl.utils import MapReduceJob, ParallelConfig, Scorer, SupervisedModel, Utility
+
+__all__ = [
+    "class_wise_shapley",
+    "CSScorer",
+]
+
+from sklearn.metrics import accuracy_score
+from tqdm import tqdm
+
+from pydvl.utils.numeric import random_powerset_group_conditional
+from pydvl.value import StoppingCriterion, ValuationResult
+
+IntArray = NDArray[np.int_]
+
+
+class CSScorer:
+    """
+    A Scorer object to be used along with the 'class_wise_shapley' function.
+    """
+
+    def __init__(self):
+        self.default = 0.0
+        self.range = (0.0, 1.0)
+        self.label: Optional[np.int_] = None
+
+    def __call__(
+        self, model: SupervisedModel, x_test: np.ndarray, y_test: np.ndarray
+    ) -> float:
+        """
+        Estimates the in and out of class accuracies and aggregated them into one float number.
+        :param model: A model to be used for predicting the labels.
+        :param x_test: The inputs to be used for measuring the accuracies. Has to match the labels.
+        :param y_test:  The labels ot be used for measuring the accuracies. It is divided further by the passed label.
+        :return: The aggregated number specified by 'in_cls_acc * exp(out_cls_acc)'
```

Comment:
Please respect the maximum column width. Sadly, black does not look at comments.

Also, the formulation is a bit unclear. What does "has to match the labels" mean? and what does it mean that y_test is "divided further"? Do you mean a stratified split?